book/accessibility.md

```diff
@@ -902,22 +926,35 @@ class Tab:
             elif is_focusable(elt):
                 self.focus_element(elt)
                 self.activate_element(elt)
-                self.set_needs_render()
                 return
             elt = elt.parent
+```
 
+Also, since any now element can be focused, we need `keypress` to
+check that an `input` element is focused before typing into it:
+
+``` {.python}
+class Tab:
     def keypress(self, char):
         if self.focus and self.focus.tag == "input":
             if not "value" in self.focus.attributes:
                 self.activate_element(self.focus)
+            # ...
 ```
 
-Note that hitting `Enter` when focused on a text entry clears the text
-entry; in most browsers, it submits the containing form instead. That
-quirk is because [our browser doesn't implement][clear-input] the
-`Backspace` key.
+I've called `activate_element` to create an empty `value` attribute.
 
-[clear-input]: forms.html#interacting-with-widgets
+Similarly, in `InputLayout` we need to only draw the cursor for
+`input` elements, even if a `button` element is focused:
```

Comment:
Ah, yeah, I could remove but I think rephrasing is better. It's basically saying: back in the day only an `<input>` element could be focused, so `if self.node.is_focused` told you whether to draw the cursor. But now that a `<button>` could also be focused, we need to do `if self.node.is_focused and self.node.tag == "input"`. (Otherwise we might draw a cursor on a focused button.)